File: backend/core/npc/npc_profile.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class NPCProfile(BaseModel):
    id: str
    name: str
    aliases: list[str] = Field(default_factory=list)
    portrait: str | None = None
    avatar: str | None = None
    chibi: str | None = None
    role: str
    faction: str | None = None
    speechStyle: str = ""
    goals: list[str] = Field(default_factory=list)
    secrets: list[str] = Field(default_factory=list)
    trustKey: str | None = None
    questIds: list[str] = Field(default_factory=list)
    knownFacts: list[str] = Field(default_factory=list)
    unlockConditions: dict[str, Any] = Field(default_factory=dict)
    promptProfile: dict[str, Any] | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)

File: frontend/src/core/npc/NPCProfile.ts
export interface NPCProfile {
  id: string;
  name: string;
  aliases: string[];
  portrait?: string | null;
  avatar?: string;
  chibi?: string;
  role: string;
  faction?: string;
  speechStyle: string;
  goals: string[];
  secrets?: string[];
  trustKey?: string;
  questIds?: string[];
  knownFacts?: string[];
  unlockConditions?: Record<string, unknown>;
  promptProfile?: {
    personality: string;
    speakingRules: string[];
    forbiddenKnowledge?: string[];
  };
  metadata?: Record<string, unknown>;
}
